source/_posts/2018-05-31-automating-your-infrastructure-deployments.markdown

```diff
@@ -0,0 +1,88 @@
+
+One of the firsts things I did in my automation journey was figuring out a way to automate the creation of cloud resources, to do that I became proficient at using some infrastructure as code tools like Terraform, AWS CloudFormation, and others.
+
+After that, since I have (or try to have) CI/CD pipelines setup in every project I am involved, my next logical step was to integrate that knowledge in infrastructure provisioning into the CI/CD pipelines. In this post I will be presenting benefits of doing that and will use an specific Terraform + CircleCI example.
+
+<!--more-->
+
+## Benefits of having automated infrastructure
+
+Let's start with the benefits of walking this path. You might be thinking: Automatically deploying infrastructure? This guy is crazy.
+
+If you are not used to this concept it might sound crazy, but in my opinion the benefits outweighs the risks. The main benefits include:
+
+- Your deploy will happen automatically.
+- You will tend to be more granular when doing changes.
+- You will be more careful when changing your infrastructure and this will almost for sure lead to less downtime.
+- You will never forget to update your branch before deploying.
+- Your development and feedback loop will be shorter, you merge your changes and everything gets applied for you. No need to wait for a "Ops" team to help you out.
+- If you need to do multiple steps for a deployment you just merge multiple PRs one after the other.
+- You can automate the testing of you infrastructure by adding some automated smoke testing after deploying (or before if you use a blue/green deployment).
+- You can invest more time in developing features.
+- All deploys can be tracked in your pipeline.
+
+## Risks of having automated infrastructure
+
+As everything in life, there are cons to this approach, but they are easy to overcome with some team training or simply by adding some security mechanisms, let's see some of the risks:
+
+- Your team must be aware of this automated process. If you merge a PR it is because you wanted to merge it. If you accidentally merge code it will get applied. In the example below I added a manual confirmation of the deployment for the `production` environment to solve this potential problem.
+- You might forget how to manually do the deploy. This is easy to solve, just manually do the deploy process every now and then.
+
+Now that we have seen the pros and cons of automating your infrastructure let's see a concrete example of a possible way of achieving that.
+
+# CI/CD with Terraform and CircleCI
+
+I created an [example repository](https://github.com/fedekau/terraform-with-circleci-example) that you can use to kickstart your project or to make your current project better with some of the ideas included there. The repository includes a detailed readme of why I took all the decisions included there.
```

Comment:
Noticed you're repeating `include` (`includes`, `included`). What do you think about something like `The repository includes a detailed readme of with the reasons why I took each decision`?